AsyncDisplayKit/ASMapNode.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+#import <MapKit/MapKit.h>
+@interface ASMapNode : ASControlNode
+- (instancetype)initWithCoordinate:(CLLocationCoordinate2D)coordinate NS_DESIGNATED_INITIALIZER;
+/**
+ This is the snapshot shot image node, this will be hidden (but not nil) when .liveMap = YES
+ */
+@property (nonatomic, readonly) ASImageNode *mapImage;
+/**
+ This is the ASDisplayNode that backs the MKMapView. This will be nil if .liveMap = NO. To access the underlying MKMapView, in order to set a delegate for example, use (MKMapView *)mapView.view;
+ */
+@property (nonatomic, readonly) ASDisplayNode *mapView;
```

Comment:
I think we indeed should expose the MKMapView, but let's call it what it is and provide a pointer directly to an MKMapView.  I don't think there is value in providing the ASDisplayNode wrapper and then requiring a typecast, especially since that would be required to set the delegate.  We also want to remind users that the MKMapView is not thread-safe, so it being exposed here directly as an MKMapView might make that a bit more clear.

As API feedback, if we were to leave this as a node, it would be important to name it with "Node" in the name (although the fact that "mapNode" would be confusingly recursive I think highlights the fact that the ASDisplayNode wrapper isn't the ideal approach).  Maybe a better name is liveMapView - but mapView seems nice too.